app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallScreen.kt

```diff
@@ -44,36 +43,43 @@ import com.wire.android.ui.theme.wireDimensions
 @Composable
 fun OngoingCallScreen(
     lifecycleOwner: LifecycleOwner = LocalLifecycleOwner.current,
-    ongoingCallViewModel: OngoingCallViewModel = hiltViewModel()
+    ongoingCallViewModel: OngoingCallViewModel = hiltViewModel(),
+    sharedCallingViewModel: SharedCallingViewModel = hiltViewModel()
 ) {
-    OngoingCallContent(ongoingCallViewModel)
-    observeScreenLifecycleChanges(lifecycleOwner, ongoingCallViewModel)
+    OngoingCallContent(sharedCallingViewModel)
+    observeScreenLifecycleChanges(lifecycleOwner, sharedCallingViewModel)
 }
 
 @OptIn(ExperimentalMaterialApi::class)
 @Composable
-private fun OngoingCallContent(ongoingCallViewModel: OngoingCallViewModel) {
+private fun OngoingCallContent(
+    sharedCallingViewModel: SharedCallingViewModel
+) {
 
     val scaffoldState = rememberBottomSheetScaffoldState()
     BottomSheetScaffold(
         topBar = {
-            //TODO to handle null name in different way
             OngoingCallTopBar(
-                ongoingCallViewModel.callEstablishedState.conversationName
-                    ?: stringResource(id = R.string.calling_label_default_caller_name)
-            ) {}
+                conversationName = when (sharedCallingViewModel.callState.conversationName) {
+                    is ConversationName.Known -> (sharedCallingViewModel.callState.conversationName as ConversationName.Known).name
+                    is ConversationName.Unknown -> stringResource(
+                        id = (sharedCallingViewModel.callState.conversationName as ConversationName.Unknown).resourceId
+                    )
+                    else -> ""
+                }
+            ) { }
         },
         sheetShape = RoundedCornerShape(topStart = dimensions().corner16x, topEnd = dimensions().corner16x),
         backgroundColor = MaterialTheme.wireColorScheme.ongoingCallBackground,
         sheetPeekHeight = dimensions().defaultSheetPeekHeight,
         scaffoldState = scaffoldState,
         sheetContent = {
-            with(ongoingCallViewModel) {
+            with(sharedCallingViewModel) {
```

Comment:
since you are wrapping it inside the with() { } how about wrap everything here so that you do not need to call for example : sharedCallingViewModel down this line